File: Python/Exercises/LeetCode/romannumeral.py
def roman_to_int(x): 
    roman_ints = {
      'I': 1, 
      'V': 5, 
      'X': 10, 
      'L': 50, 
      'C': 100, 
      'D': 500, 
      'M': 1000
      } 
    index_val = 0
    for i in range(len(x)): 
        if i > 0 and roman_ints[x[i]] > roman_ints[x[i - 1]]: 
            index_val += roman_ints[x[i]] - 2 * roman_ints[x[i - 1]] 
        else: 
            index_val += roman_ints[x[i]] 
    return index_val

print(roman_to_int('XIV'))

File: JavaScript/Exercises/LeetCode/romannumeral.js
function romanToInt(x) {
  let romanInts = {
    'I': 1,
    'V': 5,
    'X': 10,
    'L': 50,
    'C': 100,
    'D': 500,
    'M': 1000
  };


  // Single Line Solution
  // Create Array, Map Indexes, Check if Next Value > Than Current Value, TRUE: Make Negative (to Subtract in Reduce), FALSE: Keep Current Value, Reduce to Add All Values.
  return [...x].map((char, i) => romanInts[char] < romanInts[x[i + 1]] ? -Math.abs(romanInts[char]) : romanInts[char]).reduce((a, b) => a + b)

  // Initial Solution
  // let total = 0
  // for (i = 0; i < x.length; i++) {
  //   let currentNum = romanInts[x[i]];
  //   let nextNum = romanInts[x[i + 1]]

  //   if (currentNum < nextNum) {
  //     total += nextNum - currentNum;
  //     i++
  //   } else {
  //     total += currentNum
  //   }
  // }
  // return total;
}
console.log(romanToInt('MCMXCIV'));